test/e2e/specs/editor/blocks/navigation.spec.js

```diff
@@ -48,5 +61,88 @@ test.describe(
 <!-- /wp:navigation -->`
 			);
 		} );
+
+		test( 'default to my only existing menu', async ( {
+			editor,
+			page,
+			navBlockUtils,
+		} ) => {
+			const createdMenu = await navBlockUtils.createNavigationMenu( {
+				title: 'Test Menu 1',
+				content:
+					'<!-- wp:navigation-link {"label":"WordPress","type":"custom","url":"http://www.wordpress.org/","kind":"custom","isTopLevelLink":true} /-->',
+			} );
+
+			await editor.insertBlock( { name: 'core/navigation' } );
+
+			//check the block in the canvas.
+			await expect(
+				editor.canvas.locator(
+					'role=textbox[name="Navigation link text"i] >> text="WordPress"'
+				)
+			).toBeVisible();
+
+			// Check the markup of the block is correct.
+			await editor.publishPost();
+			await expect.poll( editor.getBlocks ).toMatchObject( [
+				{
+					name: 'core/navigation',
+					attributes: { ref: createdMenu.id },
+				},
+			] );
+			await page.locator( 'role=button[name="Close panel"i]' ).click();
+
+			//check the block in the frontend.
+			await page.goto( '/' );
+			await expect(
+				page.locator(
+					'role=navigation >> nth=1 >> role=link[name="WordPress"]'
```

Comment:
Any reason why we need `nth=1` here?

Reply:
Nit: In addition, it will be better if we append the `i` flag at the end of the name field to indicate case insensitiveness.

```suggestion
					'role=navigation >> role=link[name="WordPress"i]'
```

Reply:
the nth is because we have another navigation block in the footer

Reply:
But don't we only have one nav that has the "WordPress" link? As long as it's only targeting a unique element, it should be fine :).

Reply:
you are right, I added it because I debugged the selector first without the link text and I was targeting just the nav part of it. Should be good now